src/main/dynamo/QueryBuilder.ts

```diff
@@ -63,16 +47,25 @@ export class QueryBuilder<T extends TaggedModel> extends QueryExpressionBuilder<
   }
 
   async exec(): Promise<GroupedModels<T>> {
-    const results: T[] = []
-    for await (const { items } of this.executeQuery()) {
-      results.push(...items)
-    }
+    const iterator = pagedIterator<DocumentClient.QueryInput, T>(
```

Comment:
So now you can get a generic async iterator that gives you 1 "page" each time you poke it,  if you give it: 

1. Some options (`{}` blank here)
2. A function to build a DynamoDB operation given those options (e.g. `QueryInput` or `ScanInput`)
3. A function to execute the operation created in 2.